src/rpc/rpc.cc

```diff
@@ -7,25 +7,258 @@
 
 #include <dgl/runtime/container.h>
 #include <dgl/packed_func_ext.h>
+#include <dgl/zerocopy_serializer.h>
 #include "../c_api_common.h"
 
+using dgl::network::StringPrintf;
 using namespace dgl::runtime;
 
 namespace dgl {
 namespace rpc {
 
+char* SerializeRPCMeta(const RPCMessage& msg, int64_t* size) {
+  int64_t total_size = 0;
+  total_size += sizeof(msg.service_id);
+  total_size += sizeof(msg.msg_seq);
+  total_size += sizeof(msg.client_id);
+  total_size += sizeof(msg.server_id);
+  total_size += sizeof(int64_t);  // data_size
+  total_size += msg.data.size();
+  total_size += sizeof(int8_t);   // has_tensor
+  char* buffer = new char[total_size];
```

Comment:
Why not define a structure for this? it'll be much more readable.

Reply:
Yes, I will change it to dmlc serializer.

Reply:
i'm not sure about serializer. i think it's pretty simple
```C
struct struct_name {
  int64_t data_size;
  int8_t has_tensor;
  ...
};

struct_name *o = new struct_name();
o.data_size = ...;
o.has_tensor = ...;
...
```
or you can do something like this:
```C
char *mem = new char[...];
struct_name *o = static_cast<struct_name *>(mem);
...
```
it's more readable and the compiler can help check something.